src/pages/index.js

```diff
@@ -217,13 +347,39 @@ editModalCloseBtn.addEventListener("click", (evt) => {
 //Task 2/7. Form for adding a card
 
 cardModalBtn.addEventListener("click", () => {
+  cardForm.reset();
+  const inputlist = Array.from(
+    cardForm.querySelectorAll(settings.inputSelector)
+  );
+  resetValidation(cardForm, inputlist, settings); //disables the button if fields are empty
```

Comment:
This is not needed for the **card form**


* You need to **reset** the form only  in a `then` block in order not to do that immediately after `submit` or when you open the modal, because when there is an error, the modal will still be opened, and the user can try to send the data again to the server without having to type the data into the inputs once more.



```js

cardModalBtn.addEventListener("click", () => {
  openModal(cardModal);
});

```


You disable the button after submission, so it's enough